src/test/CurveCallee.t.sol

```diff
@@ -0,0 +1,321 @@
+        address who,
+        bytes calldata data
+    ) external;
+}
+
+interface Osm {
+    function read() external view returns (bytes32);
+    function kiss(address) external;
+}
+
+contract CurveCalleeTest is DSTest {
+
+    address constant hevm     = 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D;
+    address constant wstEth   = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    address constant chainlog = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
+    address constant curve    = 0xDC24316b9AE028F1497c275EB9192a3Ea0f67022;
+    address constant uniV3    = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+    uint256 constant WAD      = 1e18;
+
+    address gemJoin;
+    uint256 id;
+    address clipper;
+    CurveCallee callee;
+    uint256 tail;
+    address vat;
+    
+    function setUp() public {
+        clipper = Chainlog(chainlog).getAddress("MCD_CLIP_WSTETH_A");
+        address daiJoin = Chainlog(chainlog).getAddress("MCD_JOIN_DAI");
+        callee = new CurveCallee(curve, uniV3, daiJoin);
+        vat = Chainlog(chainlog).getAddress("MCD_VAT");
+        Vat(vat).hope(clipper);
+        tail = Clipper(clipper).tail();
+    }
+
+    function newAuction(uint256 amt) internal {
+        // wstEth._balances[address(this)] = amt;
+        Hevm(hevm).store({
+            c:   wstEth,
+            loc: keccak256(abi.encode(address(this), uint256(0))),
+            val: bytes32(amt)
+        });
+        gemJoin = Chainlog(chainlog).getAddress("MCD_JOIN_WSTETH_A");
+        Token(wstEth).approve(gemJoin, amt);
+        Join(gemJoin).join(address(this), amt);
+        (, uint256 rate, uint256 spot,,) = Vat(vat).ilks("WSTETH-A");
+        uint256 maxArt = amt * spot / rate;
+        // vat.wards[address(this)] = 1;
+        Hevm(hevm).store({
+            c:   vat,
+            loc: keccak256(abi.encode(address(this), uint256(0))),
+            val: bytes32(uint256(1))
+        });
+        Vat(vat).file("WSTETH-A", "line", type(uint256).max);
+        Vat(vat).frob({
+            i:    "WSTETH-A",
+            u:    address(this),
+            v:    address(this),
+            w:    address(this),
+            dink: int(amt),
+            dart: int(maxArt)
+        });
+        address jug = Chainlog(chainlog).getAddress("MCD_JUG");
+        Jug(jug).drip("WSTETH-A");
+        address dog = Chainlog(chainlog).getAddress("MCD_DOG");
+        id = Dog(dog).bark("WSTETH-A", address(this), address(this));
+    }
+
+    function test_baseline() public {
+        uint256 amt = 50 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0),
+            uint24(3000),
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+    }
```

Comment:
might be good to test here that the `CurveCallee` ends up with a dai balance of 0 and that the `sender` receives the `amt` joined to the vat.

Reply:
Well, that is more or less what `test_profit` (below) does. It checks that `to` (`adress(123)` in that case) has at least `minProfit` (not `amt`, which is the amount of auctioned collatéral). But good point about checking that the balance of `CurveCalle` is zéro. I will add that to `test_profit`, both for DAI and for wstETH.

Reply:
yeah i thought about that for `profit` but since the balance should be 0 whether or not there is a profit, i thought it made more sense in the baseline test

Reply:
fair enough, I will put it in the baseline test.

Reply:
done